modules/Auth/include/ReservationHandler.hpp

```diff
@@ -1,94 +1,335 @@
+
+    ///
+    /// @brief Function checks if the given \p id_token or \p parent_id_token matches the reserved token of the given \p
+    /// evse_id
+    ///
+    /// @param id_token          Id token
+    /// @param evse_id           Evse id
+    /// @param parent_id_token   Parent id token
+    /// @return The reservation id when there is a matching identifier, otherwise std::nullopt.
+    ///
+    std::optional<int32_t> matches_reserved_identifier(const std::string& id_token,
+                                                       const std::optional<uint32_t> evse_id,
+                                                       std::optional<std::string> parent_id_token);
+
+    ///
+    /// @brief Functions check if reservation at the given \p evse_id contains a parent_id
+    /// @param evse_id  Evse id
+    /// @return true if reservation for \p evse_id exists and reservation contains a parent_id
+    ///
+    bool has_reservation_parent_id(const std::optional<uint32_t> evse_id);
+
+private: // Functions
+    ///
+    /// \brief Check if there is a specific connector type in the vector.
+    /// \param evse_connectors  The vector to check for the type.
+    /// \param connector_type   The connector type to find.
+    /// \return True if the connector type is in the vector.
+    ///
+    bool has_evse_connector_type(const std::vector<Connector>& evse_connectors,
+                                 const types::evse_manager::ConnectorTypeEnum connector_type) const;
+
+    ///
+    /// \brief Check if there is at least one EVSE with the given connector type.
+    /// \param connector_type   The connector type to check.
+    /// \return True if at least one EVSE has this connector type.
+    ///
+    bool does_evse_connector_type_exist(const types::evse_manager::ConnectorTypeEnum connector_type) const;
+
+    ///
+    /// \brief Helper function to get the EVSE state, looking at the EVSE state, the state of the connectors of that
+    ///        EVSE and if there is a specific reservation for this EVSE.
+    /// \param evse_id                      The evse id to get the state from.
+    /// \param evse_specific_reservations   The evse specific reservations list to look in.
+    /// \return The `ReservationResult` to return for this specific EVSE.
+    ///
+    types::reservation::ReservationResult
+    get_evse_state(const uint32_t evse_id,
+                   const std::map<uint32_t, types::reservation::Reservation>& evse_specific_reservations);
+
+    ///
+    /// \brief Helper function to check if the connector of a specific EVSE is available.
+    /// \param evse_id          The evse id the connector belongs to.
+    /// \param connector_type   The connector type to check.
+    /// \return The `ReservationResult` to return fot his specific connector.
+    ///
+    types::reservation::ReservationResult
+    is_connector_available(const uint32_t evse_id, const types::evse_manager::ConnectorTypeEnum connector_type);
```

Comment:
These function signatures look confusing. `get_evse_state` and `is_connector_available` return a `ReservationResult`. I see that you are using the responses to return the value in `make_reservation`, but I think something more readable is possible here

Reply:
it looks better now :+1: